datafusion/core/src/physical_plan/unnest.rs

```diff
@@ -236,21 +236,21 @@ fn build_batch(
     match list_array.data_type() {
         DataType::List(_) => {
             let list_array = list_array.as_any().downcast_ref::<ListArray>().unwrap();
-            unnest_batch(batch, schema, column, &list_array)
+            unnest_batch(batch, schema, column, &list_array, list_array.values())
```

Comment:
`.values` seems to not have a common trait so it needs to be passed down here where we have the concrete types

Reply:
Yes, this makes sense